File: Array-Vector/Array-Vector_cpp/Basics/reverse_array.cpp
#include <bits/stdc++.h>
using namespace std;
int main()

{
    // Time Complexity : O(n^2)
    vector<int> arr = {1, 2, 1, 3, 4, 5};

    for (int i = 0; i < arr.size(); i++)
    {
        for (int j = i + 1; j < arr.size(); j++)
        {
            int temp = arr[i];
            arr[i] = arr[j];
            arr[j] = temp;
        }
    }

    for (int i = 0; i < arr.size(); i++)
    {
        cout << arr[i] << " ";
    }

    return 0;
}

File: Array-Vector/Array-Vector_java/Basics/reverseArray.java
public class reverseArray {
    public static void main(String[] args) {
        // Time Complexity: O(n^2)
        int[] arr = { 1, 2, 1, 3, 4, 5 };

        // Nested loops to swap elements
        for (int i = 0; i < arr.length; i++) {
            for (int j = i + 1; j < arr.length; j++) {
                // Swap elements at index i and j
                int temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
        }

        // Output: Print the modified array
        for (int i = 0; i < arr.length; i++) {
            System.out.print(arr[i] + " ");
        }
    }
}